lib/ts/TextView.h

```diff
@@ -663,7 +663,7 @@ TextView::split_prefix_at(size_t n)
   self_type zret; // default to empty return.
   if (n < this->size()) {
     zret = this->prefix(n);
-    this->remove_prefix(n + 1);
+    this->remove_prefix(std::min(n + 1, this->size()));
```

Comment:
I think I'd prefer to provide a local override of `remove_prefix` because I think the standard definition for this method is stupid - it should be completely reasonable to over specify and let the instance do the clipping. Something like
```
self_type& remove_prefix(size_t n) { this->super_type::remove_prefix(std::min(n, this->size())); return *this; }
```